packages/amplify_auth_cognito/ios/Classes/FlutterSignInResult.swift

```diff
@@ -30,14 +30,20 @@ struct FlutterSignInResult  {
     }
     
     func toJSON() -> Dictionary<String, Any> {
-      return [
-        "isSignedIn": self.isSignedIn,
-        "nextStep": [
-            "signInStep": self.signInStep,
-            "additionalInfo": self.additionalInfo,
-            "codeDeliveryDetails": self.codeDeliveryDetails
-        ]
-      ]
+      var result: Dictionary<String, Any> = ["isSignedIn": self.isSignedIn]
+      var nextStep: Dictionary<String, Any> = ["signInStep": self.signInStep]
+      
+      if (!self.codeDeliveryDetails.isEmpty) {
+        nextStep["codeDeliveryDetails"] = self.codeDeliveryDetails
+      }
+        
+      if (!self.additionalInfo.isEmpty) {
+        nextStep["additionalInfo"] = self.additionalInfo
```

Comment:
Had a similar thought here as well since it's something we ended up doing more consistently in the dart modules. But, in this case, it might be just _a little_ trickier since we're dealing with empty dictionaries rather than just nulls. I would be in favor of trying to remove empties/nulls though just because this pattern can grow unwieldy if more properties were to be added.